google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -959,3 +959,17 @@ def get_associated_experiment(self) -> Optional["aiplatform.Experiment"]:
                 location=self.location,
                 credentials=self.credentials,
             )
+
+    def delete(self, sync: bool = True) -> None:
```

Comment:
I wonder if we can add optional kwarg(s) for this method, for example, `delete_artifacts: bool = False`. Otherwise it has no difference with the higher level delete method.

Reply:
Good point. I think having a `delete_artifacts` option would be useful, however it doesn't look like the `Artifact` class has implemented delete yet: https://github.com/googleapis/python-aiplatform/blob/3d3e0aa9f0283dafe3056ed7d3067310535db48c/google/cloud/aiplatform/metadata/artifact.py#L40

I can chat with the metadata team about adding this and will follow up here.

For future reference, we can access the `ArtifactList` from a pipeline via `PipelineJob._gca_resource.job_detail.task_details`.

Reply:
Currently `Artifact` uses the high level delete method from [base.py](https://github.com/googleapis/python-aiplatform/blob/34bbd0abbaf29e644ed7703b3251f1de65bf5a86/google/cloud/aiplatform/base.py#L1200).

Good to know that we have an `ArtifactList` to track all the children artifacts! Then I think we can simply implement like this:
```
if delete_artifacts:
  for artifact in pipeline_job._gca_resource.job_detail.task_details.artifact_list:
     artifact.delete()
```